cpp/cmake/thirdparty/get_arrow.cmake

```diff
@@ -121,6 +121,7 @@ function(find_and_configure_arrow VERSION BUILD_STATIC ENABLE_S3 ENABLE_ORC ENAB
             "ARROW_GRPC_USE_SHARED ${ARROW_BUILD_SHARED}"
             "ARROW_PROTOBUF_USE_SHARED ${ARROW_BUILD_SHARED}"
             "ARROW_ZSTD_USE_SHARED ${ARROW_BUILD_SHARED}"
+            "xsimd_SOURCE BUNDLED"
```

Comment:
What is the default value here?

It looks like when the dependency source is resolved, it may have an AUTO option that attempts to find the package and falls back to the bundled copy.

https://github.com/apache/arrow/blob/b7c94e2f330b628ac3d3c92972ad691c0ec454d4/cpp/cmake_modules/ThirdpartyToolchain.cmake#L2251

https://github.com/apache/arrow/blob/b7c94e2f330b628ac3d3c92972ad691c0ec454d4/cpp/cmake_modules/ThirdpartyToolchain.cmake#L259

Reply:
`AUTO` also works on my local machine. I'm going to change this to `AUTO`.